dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/internal/DatadogContextProvider.kt

```diff
@@ -0,0 +1,101 @@
+import java.util.concurrent.TimeUnit
+
+internal class DatadogContextProvider(val coreFeature: CoreFeature) : ContextProvider {
+    override val context: DatadogContext
+        get() {
+            return DatadogContext(
+                site = coreFeature.site,
+                clientToken = coreFeature.clientToken,
+                service = coreFeature.serviceName,
+                env = coreFeature.envName,
+                version = coreFeature.packageVersion,
+                sdkVersion = coreFeature.sdkVersion,
+                source = coreFeature.sourceName,
+                time = with(coreFeature.timeProvider) {
+                    val deviceTimeMs = getDeviceTimestamp()
+                    val serverTimeMs = getServerTimestamp()
+                    TimeInfo(
+                        deviceTimeNs = TimeUnit.MILLISECONDS.toNanos(deviceTimeMs),
+                        serverTimeNs = TimeUnit.MILLISECONDS.toNanos(serverTimeMs),
+                        serverTimeOffsetNs = TimeUnit.MILLISECONDS
+                            .toNanos(serverTimeMs - deviceTimeMs),
+                        serverTimeOffsetMs = serverTimeMs - deviceTimeMs
+                    )
+                },
+                processInfo = ProcessInfo(
+                    isMainProcess = coreFeature.isMainProcess,
+                    processImportance = CoreFeature.processImportance
+                ),
+                networkInfo = with(coreFeature.networkInfoProvider.getLatestNetworkInfo()) {
+                    NetworkInfo(
+                        connectivity = connectivity.asV2(),
+                        carrier = CarrierInfo(
+                            technology = cellularTechnology,
+                            carrierName = carrierName
+                        )
+                    )
+                },
+                deviceInfo = with(coreFeature.androidInfoProvider) {
+                    DeviceInfo(
+                        deviceName = deviceName,
+                        deviceBrand = deviceBrand,
+                        deviceType = deviceType,
+                        deviceModel = deviceModel,
+                        deviceBuildId = deviceBuildId,
+                        osName = osName,
+                        osVersion = osVersion,
+                        osMajorVersion = osMajorVersion
+                    )
+                },
+                userInfo = with(coreFeature.userInfoProvider.getUserInfo()) {
+                    UserInfo(
+                        id = id,
+                        name = name,
+                        email = email,
+                        additionalProperties = additionalProperties
+                    )
+                },
+                trackingConsent = coreFeature.trackingConsentProvider.getConsent(),
+                featuresContext = emptyMap()
+            )
+        }
+
+    private fun NetworkInfoV1.Connectivity.asV2(): NetworkInfo.Connectivity {
```

Comment:
why doing the conversion and not have one single `NetworkInfo` for both v1 and v2 ?

Reply:
good point. The thing is that we indeed have some duplication for models between V1 and V2: `UserInfo` as well, for example. Eventually it will be only one, duplication is only for the V1 to V2 transition period, to have smaller changes footprint in PRs.